src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+import java.time.chrono.ChronoLocalDateTime;
+import java.time.chrono.ChronoPeriod;
+
+import java.time.temporal.ChronoField;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.Temporal;
+import java.time.temporal.TemporalAccessor;
+import java.time.temporal.TemporalField;
+import java.time.temporal.TemporalQuery;
+import java.time.temporal.TemporalUnit;
+import java.time.temporal.UnsupportedTemporalTypeException;
+import java.time.temporal.ValueRange;
+
+import java.util.Locale;
+
+/**
+ * A date in the International fixed calendar system.
+ * <p>
+ * Implements a pure International Fixed calendar (also known as the Cotsworth plan, the Eastman plan,
+ * the 13 Month calendar or the Equal Month calendar) a solar calendar proposal for calendar reform designed by
+ * Moses B. Cotsworth, who presented it in 1902.</p>
+ * <p>
+ * It provides for a year of 13 months of 28 days each, with one or two days a year belonging to no month or week.
+ * It is therefore a perennial calendar, with every date fixed always on the same weekday.
+ * Though it was never officially adopted in any country, it was the official calendar of the Eastman Kodak Company
+ * from 1928 to 1989.</p>
+ * <p>
+ * This date operates using the {@linkplain InternationalFixedChronology International fixed calendar}.
+ * This calendar system is a proposed reform calendar system, and is not in common use.
+ * The International fixed differs from the Gregorian in terms of month count and length, and the leap year rule.
+ * Dates are aligned such that {@code 0001-01-01 (International fixed)} is {@code 0001-01-01 (ISO)}.</p>
+ * <p>
+ * More information is available in the <a href='https://en.wikipedia.org/wiki/International_Fixed_Calendar'>International fixed Calendar</a> Wikipedia article.</p>
+ * <p>
+ * <h3>Implementation Requirements</h3>
+ * This class is immutable and thread-safe.
+ * <p>
+ * This class must be treated as a value type. Do not synchronize, rely on the
+ * identity hash code or use the distinction between equals() and ==.</p>
+ */
+public final class InternationalFixedDate
+        extends AbstractDate
+        implements ChronoLocalDate, Serializable {
+    /**
+     * Serialization version UID.
+     */
+    private static final long serialVersionUID = -908631218899744973L;
+    /**
+     * The days per 400 year cycle.
+     */
+    private static final int DAYS_PER_CYCLE = 146097;
+    /**
+     * Number of years in a decade.
+     */
+    private static final int YEARS_IN_DECADE = 10;
+    /**
+     * Number of years in a century.
+     */
+    private static final int YEARS_IN_CENTURY = 100;
+    /**
+     * Number of years in a millennium.
+     */
+    private static final int YEARS_IN_MILLENNIUM = 1000;
+    /**
+     * The proleptic year.
+     */
+    private final int prolepticYear;
+    /**
+     * The day of the year, it is calculated as (month - 1) * 28 + dayOfMonth + (isLeapYear() ? 1 : 0).
+     */
+    private final int dayOfYear;
```

Comment:
You're probably better off keeping month/day-of-month, to prevent checking for leap-year so often.  Since those two fields are more likely to be queried than day-of-year.